Automattic/wp-calypso
# Site Settings: Use feature checks rather than plan checks

#### Changes proposed in this Pull Request

* Replaced uses of `siteHasScanProductPurchase` with feature checks.
* Removes `siteHasScanProductPurchase` selector.

#### Testing instructions



* Navigate to /site-settings/jetpack
* Make sure you see the connection box on Atomic sites and Jetpack sites with scan feature <img width="1510" alt="image" src="https://user-images.githubusercontent.com/1398304/168914141-3a0864c8-1afd-41c9-aaa0-177c7426da7b.png">
* The changes in Navigation are only visible to sites on Jetpack below v9.9.



Related to p4TIVU-a66-p2
Depends on https://github.com/Automattic/wp-calypso/pull/63735
